tests/apps/bookinfo/productpage.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 Istio Authors
+
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+
+//       http://www.apache.org/licenses/LICENSE-2.0
+
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+)
+
+const (
+	userCookie      = "user"
+	requestIDHeader = "X-Request-ID"
+)
+
+type productPage struct {
+	Details map[string]string  `json:"details,omitempty"`
+	Reviews map[string]*review `json:"reviews,omitempty"`
+}
+
+type review struct {
+	Text   string  `json:"text,omitempty"`
+	Rating *rating `json:"rating,omitempty"`
+}
+
+type rating struct {
+	Stars int    `json:"stars,omitempty"`
+	Color string `json:"color,omitempty"`
+}
+
+func main() {
+	port := "9080"
+	http.HandleFunc("/productpage", productpageHandler)
+	http.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	log.Fatal(http.ListenAndServe(":"+port, nil))
+}
+
+func productpageHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/json")
+
+	headers := getForwardHeaders(r)
+	details := getDetails(headers)
+	reviews := getReviews(headers)
+
+	page := productPage{
+		Details: details,
+		Reviews: reviews,
+	}
+	bytes, _ := json.Marshal(page)
+
+	w.Write(bytes)
+}
+
+func getDetails(forwardHeaders http.Header) map[string]string {
+	const attempts = 1
+	const timeout = 1 * time.Second
+
+	bytes, err := doRequest("http://details.default.svc:9080/details", forwardHeaders, timeout, attempts)
```

Comment:
I'll add it to the PR in a couple of hours then you can clone my fork and build/push the new images up to docker hub.